File: C-Sharp/FerryLegacy/FerryManager.cs
using System.Collections.Generic;
using System.Linq;

namespace FerryLegacy
{
    public static class FerryManager
    {
        public static FerryJourney CreateFerryJourney(List<PortModel> ports, TimeTableEntry timetable)
        {
            if (ports == null)
                return null;

            if (timetable == null)
                return null;

            var fj = new FerryJourney
            {
                Origin = ports.Single(x => x.Id == timetable.OriginId),
                Destination = ports.Single(x => x.Id == timetable.DestinationId)
            };
            return fj;
        }

        public static void AddFerry(TimeTableEntry timetable, FerryJourney journey)
        {
            journey.Ferry = journey.Origin.GetNextAvailable(timetable.Time);
        }

        public static int GetFerryTurnaroundTime(PortModel destination)
        {
            if (destination.Id == 3)
                return 25;
            if (destination.Id == 2)
                return 20;
            return 15;
        }
    }
}

File: java/src/main/java/ferry/booking/FerryManager.java
package ferry.booking;

import java.util.List;

public class FerryManager {

    public static FerryJourney createFerryJourney(List<PortModel> ports, TimeTableEntry timetable) {
        if (ports == null) {
            return null;
        }

        if (timetable == null) {
            return null;
        }

        FerryJourney fj = new FerryJourney();
        for (PortModel port : ports) {
            if (port.id == timetable.originId) {
                fj.origin = port;
            }
            if (port.id == timetable.destinationId) {
                fj.destination = port;
            }
        }
        return fj;
    }

    public static void addFerry(TimeTableEntry timetable, FerryJourney journey) {
        journey.ferry = journey.origin.getNextAvailable(timetable.time);
    }

    public static int getFerryTurnaroundTime(PortModel destination) {
        if (destination.id == 3) {
            return 25;
        }
        if (destination.id == 2) {
            return 20;
        }
        return 15;
    }
}
